src/main/java/pico/typecheck/PICOVisitor.java

```diff
@@ -493,4 +495,31 @@ protected void checkExtendsImplements(ClassTree classTree) {
     		}
     	}
     }
+
+    /**
+     * The invoked constructor’s return type adapted to the invoking constructor’s return type must
+     * be a supertype of the invoking constructor’s return type.
+     *
+     * @param superCall the super invocation, e.g., "super()"
+     * @param errorKey the error key, e.g., "super.invocation.invalid"
+     */
+    @Override
+    protected void checkThisOrSuperConstructorCall(
+            MethodInvocationTree superCall, @CompilerMessageKey String errorKey) {
+        TreePath path = atypeFactory.getPath(superCall);
+        MethodTree enclosingMethod = TreeUtils.enclosingMethod(path);
+        AnnotatedTypeMirror superType = atypeFactory.getAnnotatedType(superCall);
+        AnnotatedExecutableType constructorType = atypeFactory.getAnnotatedType(enclosingMethod);
+        AnnotationMirror superTypeMirror = superType.getAnnotationInHierarchy(READONLY);
```

Comment:
Can you add a comment why you only compare the READONLY hierarchy?
Why don't you check whether the two ATMs are in the right relationship?
What about the FbC hierarchy?

Reply:
I have already updated the javadoc, the Initialization checker does not apply any type rules in this method, so we only check the READONLY hierarchy.

For the second question, sorry, I am not quite sure about what the relationship between two ATMs means, could you please explain more detail?